### PR TITLE
New Orchestration Module: `subPlacementAll`

### LINKED ISSUE #295: 💡 Feature Request - Subscription ID placement in Management Groups
Dear ALZ Team,

I have a number of new clients where I would like to deploy this landing zone via Bicep, I was working through the steps in my test lab but would like an easier way to define where my subscriptions go.

For example if my client has 10 subscriptions I would like to within the Bicep code define the Subscription ID and where I would like to place that within the Management group Structure (eg. xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx goes to Landing Zones - Online).  See example Screenshot : 

Screenshot 2022-07-28 081416

Would this be possible soon as I have a few clients within the next month I would like to deploy this to?

Many thanks and keep up the excellent work.

Thanks

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

Add new orchestration module called `subPlacementAll` as per request in #295 

## This PR fixes/adds/changes/removes

1. Add new orchestration module called `subPlacementAll` as per request in #295 
2. Closes #295 

### Breaking Changes

None

## Testing Evidence

Automated testing should suffice.

image

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [x] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by jtracey93** [comment]
Tests all passing and added screenshot to show sub placement working using new orchestration module (as now used in test framework on this PR). 

Markdown links failing as module doesn't exist in main branch yet but does in this PR 👍

File: docs/wiki/CustomerUsage.md [hunk 1]
```diff
@@ -24,27 +24,30 @@ module modCustomerUsageAttribution '../../CRML/customerUsageAttribution/cuaIdTen
...
+| publicIp                        | 3f85b84c-6bad-4c42-86bf-11c233241c22 |
+| resourceGroup                   | b6718c54-b49e-4748-a466-88e3d7c789c8 |
+| roleAssignments                 | 59c2ac61-cd36-413b-b999-86a3e0d958fb |
+| spokeNetworking                 | 0c428583-f2a1-4448-975c-2d6262fd193a |
+| subscriptionPlacement           | 3dfa9e81-f0cf-4b25-858e-167937fd380b |
+| virtualNetworkPeer              | ab8e3b12-b0fa-40aa-8630-e3f7699e2142 |
+| vwanConnectivity                | 7f94f23b-7a59-4a5c-9a8d-2a253a566f61 |
+| vnetPeeringVwan                 | 7b5e6db2-1e8c-4b01-8eee-e1830073a63d |
+| privateDnsZones                 | 981733dd-3195-4fda-a4ee-605ab959edb6 |
+| hubSpoke - Orchestration        | 50ad3b1a-f72c-4de4-8293-8a6399991beb |
```
Comment (jfaurskov) [inline docs/wiki/CustomerUsage.md:51]: Not directly related to this PR, but this module seems to be mssing.

**Review by jfaurskov — CHANGES_REQUESTED** [review]
Nice work @jtracey93. A couple of comments. Also I started the mc validate pipeline too.

File: tests/pipelines/bicep-build-to-validate.yml [hunk 2]
```diff
@@ -177,7 +178,7 @@ jobs:
     inputs:
       targetType: 'inline'
       script: |
-        az deployment mg create --template-file infra-as-code/bicep/modules/subscriptionPlacement/subscriptionPlacement.bicep --parameters @infra-as-code/bicep/modules/subscriptionPlacement/parameters/subscriptionPlacement.parameters.min.json parTargetManagementGroupId=$(ManagementGroupPrefix)-platform-connectivity parSubscriptionIds='["$(subscriptionId)"]' --location $(Location) --management-group-id $(ManagementGroupPrefix)
```
[the remark this answers is not in the document]
Comment (jtracey93) [reply tests/pipelines/bicep-build-to-validate.yml:180]: So, my thinking here is that as the orchestration module `subPlacementAll` calls and uses the `subscriptionPlacement` module. We are technically testing deployment of both in one go 👍

As all the `subPlacementAll` orchestration module is doing is taking parameters of sub IDs and passing them to a instance call of the `subscriptionPlacement` module at a specified scope.

File: docs/wiki/CustomerUsage.md (hunk 1 above)
Comment (jtracey93) [reply docs/wiki/CustomerUsage.md:51]: Nope its there - https://github.com/Azure/ALZ-Bicep/pull/298/files#diff-fe19a3b373a175486c649ce7646942d22c0b7303568cc4cf5c75a7dea24fa39cR53
image

Comment (jfaurskov) [reply docs/wiki/CustomerUsage.md:51]: I meant the one in line 51 :-) the hubSpoke

**Review by jfaurskov — APPROVED** [review]
LGTM

**Pull request merged**